frontend/src/app/app.component.ts

```diff
@@ -10,13 +11,18 @@ import { AuthService } from './services/auth/auth.service';
 export class AppComponent implements OnInit {
   public title: string = 'online-fishing-store';
 
-  constructor(private authService: AuthService){}
+  constructor(
+    private authService: AuthService,
+    private productService: ProductService,
+    ){}
 
   ngOnInit(): void {
     const token: string = localStorage.getItem(LocalStorageKey.AUTH_TOKEN);
 
     if(token) {
       this.authService.setToken(token);
     }
+
+    this.productService.getProductsFromServer().subscribe();
```

Comment:
1) I can't understand for what you need products in root component which you don't even getting and using
2) getProductsFromServer - It si bad name. Rename to "getProducts"